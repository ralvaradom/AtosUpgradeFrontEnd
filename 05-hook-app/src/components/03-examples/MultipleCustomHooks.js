import React from 'react';
import useCounter from '../../hooks/useCounter';
import useFetch from '../../hooks/useFetch';

const MultipleCustomHooks = () => {

	const { counter, increment } = useCounter();

	const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
	const { author, quote } = !!data && data[0];
	
  	return (
		  <>
			<h1>BrakingBad Quotes</h1>
			<hr />

			{
				loading
					?
						(
							<div className="alert alert-info text-center">
								Loading...
							</div>
						)
					:
						(
							<blockquote className="blockquote text-right">
								<p class="mb-0">{quote}</p>
								<footer className="blockquote-footer">{author}</footer>
							</blockquote>
						)
			}

			<button className="btn btn-primary" onClick={increment}>
				Next quote
			</button>
		  </>
	  );
};

export default MultipleCustomHooks;
