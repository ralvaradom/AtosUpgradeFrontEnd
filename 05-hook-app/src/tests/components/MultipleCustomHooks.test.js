import React from 'react';
import { shallow } from 'enzyme';
import MultipleCustomHooks from '../../components/03-examples/MultipleCustomHooks';
import useFetch from '../../hooks/useFetch';

jest.mock('../../hooks/useFetch');

describe('Pruebas en <MultileCustomHooks />', () => {

	test('debe mostrarse correctamente', () => {

		useFetch.mockReturnValue({
			data: null,
			loading: true,
			error: null
		});

		const wrapper = shallow(<MultipleCustomHooks />);
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de mostrar la informacion', () => {

		useFetch.mockReturnValue({
			data: [{
				author: 'Fernando',
				quote: 'hola mundo'
			}],
			loading: false,
			error: null
		});

		const wrapper = shallow(<MultipleCustomHooks />);

		expect(wrapper.find('.alert').exists()).toBe(false);
		expect(wrapper.find('p').text().trim()).toBe('hola mundo'); 
		expect(wrapper.find('footer').text().trim()).toBe('Fernando');

	});
});
