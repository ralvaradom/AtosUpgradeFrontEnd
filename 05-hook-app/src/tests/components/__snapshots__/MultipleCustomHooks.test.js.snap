// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultileCustomHooks /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    BrakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Next quote
  </button>
</Fragment>
`;
